### PR TITLE
Simplify ValueObject::GetQualifiedRepresentationIfAvailable().

### PR DESCRIPTION
**Pull request by adrian-prantl**
I received a couple of nullptr-deref crash reports with no line numbers in this function. The way the function was written it was a bit diffucult to keep track of when result_sp could be null, so this patch simplifies the function to make it more obvious when a nullptr can be contained in the variable.

### CODE REVIEW

**Comment by adrian-prantl** [comment]
rdar://118018930

**Review by Michael137 — APPROVED** [review]
LGTM

File: lldb/source/Core/ValueObject.cpp [hunk 1]
```diff
@@ -2594,34 +2594,30 @@ ValueObjectSP ValueObject::CreateConstantValue(ConstString name) {
...
-    if (!result_sp->IsSynthetic()) {
-      if (result_sp->GetSyntheticValue())
-        result_sp = result_sp->GetSyntheticValue();
-    }
-  } else {
-    if (result_sp->IsSynthetic()) {
-      if (result_sp->GetNonSyntheticValue())
-        result_sp = result_sp->GetNonSyntheticValue();
-    }
+  bool is_synthetic = result_sp->IsSynthetic();
```
Comment (bulbazord) [inline lldb/source/Core/ValueObject.cpp:2613]: const-ify?

**Review by bulbazord — APPROVED** [review]
LGTM

**Pull request merged**